extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/SessionCachingMetadataDomainRepository.java

```diff
@@ -144,9 +158,16 @@ public SessionCachingMetadataDomainRepository( final IMetadataDomainRepository d
     if ( systemConfig != null ) {
       String enableDomainIdCache = systemConfig.getProperty( "system.enableDomainIdCache" );
       domainIdsCacheEnabled = ( enableDomainIdCache == null ) || Boolean.valueOf( enableDomainIdCache );
+      numberOfThreads = getNumberOfThreads( systemConfig );
     }
   }
 
```

Comment:
TODO i need to add javadoc comment for function

Reply:
see commit: https://github.com/pentaho/pentaho-platform/pull/5124/commits/2277f57c5f89124e2e5b11b6591f46694b98350d